src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int anInt = random.nextInt(5);
+        return switch (anInt) {
+            case 1 -> new Circle(Colors.WHITE.name(), 10);
+            case 2 -> new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                    random.nextInt(10), random.nextInt(10));
+            case 3 -> new Rectangle(colorSupplier.getRandomColor(), random.nextInt(10),
+                    random.nextInt(10));
+            case 4 -> new RightTriangle(colorSupplier.getRandomColor(), random.nextInt(10),
+                    random.nextInt(10));
+            default -> new Square(colorSupplier.getRandomColor(), random.nextInt(10),
+                    random.nextInt(10));
+        };
+    }
```

Comment:
All magic numbers in your code should be constants. Instead of using '5' directly in `random.nextInt(5)`, you should define it as a constant, for example, `private static final int FIGURE_COUNT = 5;` and then use `FIGURE_COUNT` in your method.